crates/sparrow-session/src/session.rs

```diff
@@ -34,17 +35,37 @@ pub struct Session {
     udfs: HashMap<Uuid, Arc<dyn Udf>>,
     object_store_registry: Arc<ObjectStoreRegistry>,
     rt: tokio::runtime::Runtime,
+    /// Url to prepare files to.
+    ///
+    /// May be a path to a temporary directory, remote object store, etc.
+    /// Note: Creating in the Session ensures temporary files won't be dropped
+    /// until the session is closed.
+    prepare_prefix: ObjectStoreUrl,
+    /// Temporary directory for preparing files to in the local case.
+    ///
+    /// Stored in the session to ensure it persists until the Session is dropped.
+    _temp_dir: tempfile::TempDir,
 }
 
 impl Default for Session {
     fn default() -> Self {
+        // TODO: Support object stores
+        // Likely will need the option to pass in the destination url when executing the
+        // query or creating a table.
+        let temp_dir = tempfile::tempdir().expect("create temp dir");
+        let mut url = "file://".to_owned();
+        url.push_str(temp_dir.path().to_str().expect("valid string"));
```

Comment:
I suspect this will be a problem on windows -- paths will have `\` but URLs will want `/`.

You can likely do something with https://docs.rs/object_store/latest/object_store/path/struct.Path.html#method.from_filesystem_path to convert the filesystem path to an `object_store::path::Path` which is what our `ObjectStoreUrl` is based on. Maybe even just add a `ObjectStoreUrl::from_filesystem_dir(...)` or something?

Reply:
`from_filesystem_dir`: "Convert a filesystem path to a [Path](https://docs.rs/object_store/latest/object_store/path/struct.Path.html) relative to the filesystem root"

I'm not sure how this would help. I can convert the tempdir relative to the root, but then I still need the `file://` prefix for it to be formatted as a URL, unless I misunderstand.

Reply:
The `object_store::path::Path` is actually a URL (just poorly named). So I believe that it would convert `C:\\path\to\tempfile` into `file://C:/path/to/tempfile`.